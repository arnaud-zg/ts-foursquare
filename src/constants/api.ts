export const CLIENT_ID_KEY = 'apiClientId'
export const CLIENT_SECRET_KEY = 'apiClientSecret'

export interface ICredentials {
  [CLIENT_ID_KEY]: string
  [CLIENT_SECRET_KEY]: string
}

export enum EApiDefaultParameters {
  VERSION = '20190425',
}
