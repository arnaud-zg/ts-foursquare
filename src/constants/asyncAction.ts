import { getType } from 'typesafe-actions'
import {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
} from '../DEPRECATED_actions'

interface IAsyncActionNameMapping {
  [actionType: string]: string
}

export const ASYNC_ACTION_NAME_MAPPING: IAsyncActionNameMapping = {
  // getVenuesCategoriesAsync
  [getType(getVenuesCategoriesAsync.request)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.success)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.failure)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.cancel)]: 'getVenuesCategoriesAsync',
  // getVenuesExploreAsync
  [getType(getVenuesExploreAsync.request)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.success)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.failure)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.cancel)]: 'getVenuesExploreAsync',
  // getVenuesLikesAsync
  [getType(getVenuesLikesAsync.request)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.success)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.failure)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.cancel)]: 'getVenuesLikesAsync',
  // getVenuesListedAsync
  [getType(getVenuesListedAsync.request)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.success)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.failure)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.cancel)]: 'getVenuesListedAsync',
  // getVenuesNextVenuesAsync
  [getType(getVenuesNextVenuesAsync.request)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.success)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.failure)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.cancel)]: 'getVenuesNextVenuesAsync',
  // getVenuesSearchAsync
  [getType(getVenuesSearchAsync.request)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.success)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.failure)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.cancel)]: 'getVenuesSearchAsync',
  // getVenuesSimilarAsync
  [getType(getVenuesSimilarAsync.request)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.success)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.failure)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.cancel)]: 'getVenuesSimilarAsync',
  // getVenuesSuggestCompletionAsync
  [getType(
    getVenuesSuggestCompletionAsync.request
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.success
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.failure
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.cancel
  )]: 'getVenuesSuggestCompletionAsync',
  // getVenuesTrendingAsync
  [getType(getVenuesTrendingAsync.request)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.success)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.failure)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.cancel)]: 'getVenuesTrendingAsync',
}
