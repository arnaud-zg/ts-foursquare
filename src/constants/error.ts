export enum ECommonError {
  RESPONSE_NOT_VALID = 'Response is not valid',
}
