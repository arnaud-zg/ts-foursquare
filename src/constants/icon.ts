export enum EIconSize {
  S = '32',
  M = '44',
  L = '64',
  XL = '88',
}
