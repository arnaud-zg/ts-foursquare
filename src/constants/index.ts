export const PROJECT_NAME = 'ts-foursquare'
