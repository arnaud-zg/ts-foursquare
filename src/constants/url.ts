export enum EUrlSeparator {
  LOCATION_SEARCH_SEPARATOR = '?',
  QUERY_STRING_SEPARATOR = '&',
}
