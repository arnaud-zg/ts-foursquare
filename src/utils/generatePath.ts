import pathToRegexp from 'path-to-regexp'

const compilePath = (path: string) => pathToRegexp.compile(path)

/**
 * Public API for generating a URL pathname from a path and parameters.
 * https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/generatePath.js#L20
 */
export const generatePath = (path = '/', params = {}) =>
  path === '/' ? path : compilePath(path)(params)
