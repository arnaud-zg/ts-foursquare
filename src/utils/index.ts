export * from './icon'
export * from './url'
