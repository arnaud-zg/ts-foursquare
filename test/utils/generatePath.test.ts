import { EApiPathnames } from '../../src/constants/api'
import { generatePath } from '../../src/utils/generatePath'

describe('utils/icon/generatePath', () => {
  it('should get default url pathname', () => {
    expect(generatePath()).toMatchSnapshot()
  })

  it('should get url pathname', () => {
    expect(generatePath(EApiPathnames.VENUES_EXPLORE)).toMatchSnapshot()
    expect(generatePath(EApiPathnames.VENUES_EXPLORE, {})).toMatchSnapshot()
  })

  it('should get url pathname with given parameters', () => {
    expect(
      generatePath(EApiPathnames.VENUES_LIKES, {
        venueId: '3fd66200f964a5209beb1ee3',
      })
    ).toMatchSnapshot()
  })
})
