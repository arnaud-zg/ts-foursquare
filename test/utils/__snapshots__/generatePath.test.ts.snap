// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/icon/generatePath should get default url pathname 1`] = `"/"`;

exports[`utils/icon/generatePath should get url pathname 1`] = `"/v2/venues/explore"`;

exports[`utils/icon/generatePath should get url pathname 2`] = `"/v2/venues/explore"`;

exports[`utils/icon/generatePath should get url pathname with given parameters 1`] = `"/v2/venues/3fd66200f964a5209beb1ee3/likes"`;
